src/App.tsx

```diff
@@ -1,36 +1,45 @@
-import { Flex, useDisclosure, Text, Box } from '@chakra-ui/react'
+import { Box, Flex, Text, useDisclosure } from '@chakra-ui/react'
 import Overview from './views/Overview'
 import { ViewState } from './constants/enums'
 import { useApplicationContext } from './contexts/applicationContext'
 import Macroview from './views/Macroview'
 import { useEffect } from 'react'
-import SettingsModal from './views/SettingsModal'
 import data from '@emoji-mart/data'
 import { init } from 'emoji-mart'
 import './App.css'
 import { MacroProvider } from './contexts/macroContext'
+import MacroSettingsModal from './views/MacroSettingsModal'
+import SettingsModal from './views/SettingsModal'
 
 function App() {
   const { viewState, initComplete } = useApplicationContext()
-  const { isOpen, onOpen, onClose } = useDisclosure()
+  const {
+    isOpen: isOpenSettings,
+    onOpen: onOpenSettings,
+    onClose: onCloseSettings
+  } = useDisclosure()
+  const {
+    isOpen: isOpenMacroSettings,
+    onOpen: onOpenMacroSettings,
+    onClose: onCloseMacroSettings
+  } = useDisclosure()
```

Comment:
Just wondering, and this will be in the list of things to look at next PR, but why not have the modal for the individual macro settings be inside the `Macroview` component?

Then you wouldn't have to give aliases to both disclosures

Reply:
Good note, will addresss in next PRs